From ebc890d3885c1cdc4f02eb72ed5214eb97804232 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Wed, 10 Feb 2016 15:58:00 -0800
Subject: [PATCH] laird: PIO: be sure to clear PDs on output also

If an output, the function clears the PUs. Seems to me we should clear any
PDs also. This is important for a few of our pins, so do it.

Bug: 8628
---
 driver/at91_pio.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/driver/at91_pio.c b/driver/at91_pio.c
index 9e05433..57b4d07 100644
--- a/driver/at91_pio.c
+++ b/driver/at91_pio.c
@@ -217,6 +217,9 @@ int pio_set_gpio_output(unsigned pin, int value)
 
 	write_pio(pio, PIO_IDR, mask);
 	write_pio(pio, PIO_PPUDR, mask);
+#ifdef CONFIG_HAS_PIO3
+	write_pio(pio, PIO_PPDDR, mask);
+#endif
 	write_pio(pio, (value ? PIO_SODR : PIO_CODR), mask);
 	write_pio(pio, PIO_OER, mask);
 	write_pio(pio, PIO_PER, mask);
-- 
1.9.1

